components/TextFields/src/ContainedInputView/MDCBaseTextField.m

```diff
@@ -60,11 +64,39 @@ - (void)setUpLayoutDirection {
 
 #pragma mark UIView Overrides
 
+- (void)layoutSubviews {
+  [self preLayoutSubviews];
+  [super layoutSubviews];
+  [self postLayoutSubviews];
+}
+
 - (void)traitCollectionDidChange:(UITraitCollection *)previousTraitCollection {
   [super traitCollectionDidChange:previousTraitCollection];
   [self setUpLayoutDirection];
 }
 
+#pragma mark Layout
+
+- (void)preLayoutSubviews {
+  CGSize fittingSize = CGSizeMake(CGRectGetWidth(self.frame), CGFLOAT_MAX);
+  self.layout = [self calculateLayoutWithTextFieldSize:fittingSize];
```

Comment:
Why does this need to be called before `super layoutSubviews`?

Reply:
I added documentation above `-preLayoutSubviews`.